### ISSUE TITLE
Change the signature of internal registering gluster-exporter function 

### ISSUE THREAD

**Issue by aruniiird** [gluster/gluster-prometheus#100]
Current registering function has a signature as follows,
`func () error`
During the implementation side, we tend to use **global** variables (which are initialized in the `main` function).
This issue is created to standardize the registering function in such a way that we avoid any use of global variables and instead we pass the needed variables as argument into the function.

For example, it can be changed to accept a `GInterface` object (as most of the time we use a GInterface object  to get `LocalPeerID` etc)
`func (glusterutils.GInterface) error`

or if you require / need config details also in the registering function, the signature can be,
`func (glusterutils.GInterface, *conf.Config) error`

We may have to asses the existing code and see what all global variables are being used and generalize the function signature.

**Comment by aruniiird** [comment]
@shtripat @aravindavk , if my wordings or acronyms are not right, please feel free to (propose-)change/edit the contents... (=

**Comment by shtripat** [comment]
@aruniiird makes sense and later this interface can expose a channel as explained in https://github.com/gluster/gluster-prometheus/issues/125 which would get details like local peer id, volume info, leader info etc. @aravindavk does this make sense?
